<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mainWindow</class>
 <widget class="QMainWindow" name="mainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>619</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Voronoi Viewer 3D</string>
  </property>
  <property name="windowIcon">
   <iconset resource="v_resources.qrc">
    <normaloff>:/v_res/Resources/icons/voro_icon.png</normaloff>:/v_res/Resources/icons/voro_icon.png</iconset>
  </property>
  <property name="locale">
   <locale language="Ukrainian" country="Ukraine"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="windowGridLayout">
    <property name="bottomMargin">
     <number>6</number>
    </property>
    <item row="0" column="0">
     <widget class="QTabWidget" name="mainContent">
      <property name="tabPosition">
       <enum>QTabWidget::North</enum>
      </property>
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="elideMode">
       <enum>Qt::ElideNone</enum>
      </property>
      <property name="documentMode">
       <bool>false</bool>
      </property>
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
      <property name="movable">
       <bool>true</bool>
      </property>
      <widget class="VCanvas" name="initalContentWrapper">
       <attribute name="title">
        <string>Початкова сторінка</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="initalImageLabel">
          <property name="text">
           <string/>
          </property>
          <property name="pixmap">
           <pixmap resource="v_resources.qrc">:/v_res/Resources/inital.png</pixmap>
          </property>
          <property name="scaledContents">
           <bool>false</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="file">
    <property name="title">
     <string>Файл</string>
    </property>
    <addaction name="newDiagramAction"/>
    <addaction name="separator"/>
    <addaction name="exit"/>
   </widget>
   <widget class="QMenu" name="edit">
    <property name="title">
     <string>Правка</string>
    </property>
    <addaction name="editParticlesAction"/>
   </widget>
   <widget class="QMenu" name="view">
    <property name="title">
     <string>Вид</string>
    </property>
    <addaction name="toggleParticleEditorAction"/>
    <addaction name="toggleCellsViewerAction"/>
    <addaction name="separator"/>
    <addaction name="toggleParticlesNumbersVisibilityAction"/>
    <addaction name="toggleAxesVisibilityAction"/>
    <addaction name="separator"/>
    <addaction name="restoreCameraDefaultViewAction"/>
    <addaction name="separator"/>
    <addaction name="toggleFullscreenAction"/>
   </widget>
   <addaction name="file"/>
   <addaction name="edit"/>
   <addaction name="view"/>
  </widget>
  <widget class="QDockWidget" name="particlesEditor">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="windowIcon">
    <iconset resource="v_resources.qrc">
     <normaloff>:/v_res/Resources/icons/note_edit.png</normaloff>:/v_res/Resources/icons/note_edit.png</iconset>
   </property>
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>Редактор частинок</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="particlesEditorWrapper">
    <layout class="QGridLayout" name="gridLayout">
     <property name="leftMargin">
      <number>6</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>6</number>
     </property>
     <property name="horizontalSpacing">
      <number>0</number>
     </property>
     <property name="verticalSpacing">
      <number>1</number>
     </property>
     <item row="0" column="0">
      <widget class="QToolButton" name="addParticle">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="v_resources.qrc">
         <normaloff>:/v_res/Resources/icons/plus_32.png</normaloff>:/v_res/Resources/icons/plus_32.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
       <property name="checkable">
        <bool>false</bool>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
       <property name="autoRepeat">
        <bool>false</bool>
       </property>
       <property name="popupMode">
        <enum>QToolButton::DelayedPopup</enum>
       </property>
       <property name="autoRaise">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QToolButton" name="deleteParticle">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="v_resources.qrc">
         <normaloff>:/v_res/Resources/icons/delete_32.png</normaloff>:/v_res/Resources/icons/delete_32.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
       <property name="popupMode">
        <enum>QToolButton::DelayedPopup</enum>
       </property>
       <property name="toolButtonStyle">
        <enum>Qt::ToolButtonIconOnly</enum>
       </property>
       <property name="autoRaise">
        <bool>true</bool>
       </property>
       <property name="arrowType">
        <enum>Qt::NoArrow</enum>
       </property>
      </widget>
     </item>
     <item row="1" column="0" colspan="3">
      <widget class="VParticlesEditTable" name="particlesEditorTable">
       <property name="gridStyle">
        <enum>Qt::SolidLine</enum>
       </property>
       <property name="sortingEnabled">
        <bool>false</bool>
       </property>
       <attribute name="horizontalHeaderCascadingSectionResizes">
        <bool>false</bool>
       </attribute>
       <attribute name="horizontalHeaderDefaultSectionSize">
        <number>50</number>
       </attribute>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="cellsViewer">
   <property name="acceptDrops">
    <bool>false</bool>
   </property>
   <property name="windowIcon">
    <iconset resource="v_resources.qrc">
     <normaloff>:/v_res/Resources/icons/list.png</normaloff>:/v_res/Resources/icons/list.png</iconset>
   </property>
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>Комірки</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="cellsViewerWrapper">
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="leftMargin">
      <number>6</number>
     </property>
     <property name="topMargin">
      <number>3</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>6</number>
     </property>
     <item>
      <widget class="VCellsViewerTable" name="cellsViewerTable">
       <property name="contextMenuPolicy">
        <enum>Qt::DefaultContextMenu</enum>
       </property>
       <column>
        <property name="text">
         <string>Колір</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Видимість</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Інформ.</string>
        </property>
       </column>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="newDiagramAction">
   <property name="icon">
    <iconset resource="v_resources.qrc">
     <normaloff>:/v_res/Resources/icons/document_16.png</normaloff>:/v_res/Resources/icons/document_16.png</iconset>
   </property>
   <property name="text">
    <string>Нова діаграма</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
   <property name="menuRole">
    <enum>QAction::ApplicationSpecificRole</enum>
   </property>
  </action>
  <action name="openAction">
   <property name="text">
    <string>Открыть</string>
   </property>
  </action>
  <action name="newWindowAction">
   <property name="text">
    <string>Новое окно</string>
   </property>
  </action>
  <action name="saveAction">
   <property name="text">
    <string>Сохранить</string>
   </property>
  </action>
  <action name="saveAsAction">
   <property name="text">
    <string>Сохранить как ...</string>
   </property>
  </action>
  <action name="exit">
   <property name="text">
    <string>Вийти</string>
   </property>
   <property name="menuRole">
    <enum>QAction::QuitRole</enum>
   </property>
  </action>
  <action name="editParticlesAction">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="v_resources.qrc">
     <normaloff>:/v_res/Resources/icons/application_edit.png</normaloff>:/v_res/Resources/icons/application_edit.png</iconset>
   </property>
   <property name="text">
    <string>Вхідні данні</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="toggleParticleEditorAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="v_resources.qrc">
     <normaloff>:/v_res/Resources/icons/note_edit.png</normaloff>:/v_res/Resources/icons/note_edit.png</iconset>
   </property>
   <property name="text">
    <string>Редактор вхідних данних</string>
   </property>
  </action>
  <action name="toggleCellsViewerAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="v_resources.qrc">
     <normaloff>:/v_res/Resources/icons/list.png</normaloff>:/v_res/Resources/icons/list.png</iconset>
   </property>
   <property name="text">
    <string>Комірки</string>
   </property>
  </action>
  <action name="getDiagramInfoAction">
   <property name="text">
    <string>Информация о диаграме</string>
   </property>
  </action>
  <action name="getCellInfoAction">
   <property name="text">
    <string>Информация о ячейке</string>
   </property>
  </action>
  <action name="toggleParticlesNumbersVisibilityAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Номера частинок</string>
   </property>
  </action>
  <action name="restoreCameraDefaultViewAction">
   <property name="icon">
    <iconset resource="v_resources.qrc">
     <normaloff>:/v_res/Resources/icons/video.png</normaloff>:/v_res/Resources/icons/video.png</iconset>
   </property>
   <property name="text">
    <string>Положення камери за умовчанням</string>
   </property>
   <property name="menuRole">
    <enum>QAction::ApplicationSpecificRole</enum>
   </property>
  </action>
  <action name="toggleAxesVisibilityAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Координатні вісі</string>
   </property>
  </action>
  <action name="toggleFullscreenAction">
   <property name="text">
    <string>Повноекранний режим</string>
   </property>
   <property name="shortcut">
    <string>F11</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>VCanvas</class>
   <extends>QWidget</extends>
   <header>VCanvas.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>VCellsViewerTable</class>
   <extends>QTableWidget</extends>
   <header>VCellsViewerTable.h</header>
   <slots>
    <signal>cellColorChanged(int,QColor)</signal>
    <signal>cellVisibilityToggled(int)</signal>
    <signal>cellInfoCalled(int)</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>VParticlesEditTable</class>
   <extends>QTableWidget</extends>
   <header>VParticlesEditTable.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="v_resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>newDiagramAction</sender>
   <signal>triggered()</signal>
   <receiver>mainWindow</receiver>
   <slot>newFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toggleParticleEditorAction</sender>
   <signal>triggered(bool)</signal>
   <receiver>particlesEditor</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>85</x>
     <y>315</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>particlesEditor</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>toggleParticleEditorAction</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>315</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addParticle</sender>
   <signal>clicked()</signal>
   <receiver>particlesEditorTable</receiver>
   <slot>addRow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>12</x>
     <y>91</y>
    </hint>
    <hint type="destinationlabel">
     <x>28</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>deleteParticle</sender>
   <signal>clicked()</signal>
   <receiver>particlesEditorTable</receiver>
   <slot>deleteRow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>55</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>66</x>
     <y>143</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>particlesEditorTable</sender>
   <signal>cellChanged(int,int)</signal>
   <receiver>particlesEditorTable</receiver>
   <slot>checkRow(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>272</y>
    </hint>
    <hint type="destinationlabel">
     <x>101</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>particlesEditorTable</sender>
   <signal>particleChanged(int,double,double,double)</signal>
   <receiver>mainWindow</receiver>
   <slot>updateParticle(int,double,double,double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>78</x>
     <y>331</y>
    </hint>
    <hint type="destinationlabel">
     <x>570</x>
     <y>-12</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>particlesEditorTable</sender>
   <signal>particleDeleted(int)</signal>
   <receiver>mainWindow</receiver>
   <slot>deleteParticle(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>105</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>417</x>
     <y>-19</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editParticlesAction</sender>
   <signal>triggered()</signal>
   <receiver>mainWindow</receiver>
   <slot>editParticles()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toggleCellsViewerAction</sender>
   <signal>triggered(bool)</signal>
   <receiver>cellsViewer</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>86</x>
     <y>487</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cellsViewer</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>toggleCellsViewerAction</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>130</x>
     <y>471</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cellsViewerTable</sender>
   <signal>cellVisibilityToggled(int)</signal>
   <receiver>mainWindow</receiver>
   <slot>toggleCellVisibility(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>228</x>
     <y>404</y>
    </hint>
    <hint type="destinationlabel">
     <x>272</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cellsViewerTable</sender>
   <signal>cellColorChanged(int,QColor)</signal>
   <receiver>mainWindow</receiver>
   <slot>changeCellColor(int,QColor)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>151</x>
     <y>450</y>
    </hint>
    <hint type="destinationlabel">
     <x>269</x>
     <y>441</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cellsViewerTable</sender>
   <signal>cellInfoCalled(int)</signal>
   <receiver>mainWindow</receiver>
   <slot>showCellInfo(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>126</x>
     <y>497</y>
    </hint>
    <hint type="destinationlabel">
     <x>270</x>
     <y>521</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mainContent</sender>
   <signal>tabCloseRequested(int)</signal>
   <receiver>mainWindow</receiver>
   <slot>closeContentTab(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>543</x>
     <y>266</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>137</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toggleParticlesNumbersVisibilityAction</sender>
   <signal>triggered()</signal>
   <receiver>mainWindow</receiver>
   <slot>toggleParticlesNumbersVisibility()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>restoreCameraDefaultViewAction</sender>
   <signal>triggered()</signal>
   <receiver>mainWindow</receiver>
   <slot>restoreDefaultView()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exit</sender>
   <signal>triggered()</signal>
   <receiver>mainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toggleFullscreenAction</sender>
   <signal>triggered(bool)</signal>
   <receiver>mainWindow</receiver>
   <slot>showFullScreen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toggleAxesVisibilityAction</sender>
   <signal>triggered()</signal>
   <receiver>mainWindow</receiver>
   <slot>toggleAxesVisibility()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>newFile()</slot>
  <slot>createNewFile(QString,int,VPoint,VPoint)</slot>
  <slot>toggleParticlesEditor()</slot>
  <slot>updateParticle(int,double,double,double)</slot>
  <slot>deleteParticle(int)</slot>
  <slot>editParticles()</slot>
  <slot>changeCellColor(int,QColor)</slot>
  <slot>toggleCellVisibility(int)</slot>
  <slot>showCellInfo(int)</slot>
  <slot>closeContentTab(int)</slot>
  <slot>toggleParticlesNumbersVisibility()</slot>
  <slot>restoreDefaultView()</slot>
  <slot>toggleAxesVisibility()</slot>
 </slots>
</ui>
